llvm/lib/Transforms/Scalar/SROA.cpp

```diff
@@ -5045,10 +5045,20 @@ static void insertNewDbgInst(DIBuilder &DIB, DbgVariableRecord *Orig,
                                                    BeforeInst->getIterator());
     return;
   }
+
+  if (Orig->isDbgValue()) {
```

Comment:
I'll comment this here in case it's relevant: this patch is adding support for `#dbg_value` to SROA, but doesn't add it for `llvm.dbg.value`. Maybe that's fine, since we're moving from intrinsics to records and already use records by default, so we don't need to include intrinsics in every new patch; just noting that if you intend to make use of debug intrinsics downstream, you'll need to add support for `DbgValueInst` in the same way you have for debug value records.

Reply:
You are correct on the semantic difference, but as far as I understand, the `DbgValueInst` will be converted to a `#dbg_value` before SROA is run, so I don't think we do not need to add support for `DbgValueInst`